pkgs/tools/misc/vsh/default.nix

```
@@ -0,0 +1,28 @@
+{ buildGoModule, fetchFromGitHub, lib }:
+
+buildGoModule rec {
+  pname = "vsh";
+  version = "0.7.2";
+
+  src = fetchFromGitHub {
+    owner = "fishi0x01";
+    repo = "vsh";
+    rev = "v${version}";
+    sha256 = "0k9bbfzqhijzimng8plk2xx9h37h7d2wj8g3plvvs3wrf7pmwxs7";
+  };
+
+  # vendor dir is already part of repository
+  vendorSha256 = null;
+
+  # make sure version gets set at compile time
+  buildPhase = ''
+    go install -ldflags "-X main.vshVersion=v${version}"
+  '';
```

```suggestion
  buildFlagsArray = [ "-ldflags=-s -w -X main.vshVersion=v${version}" ];
```
According to other go packages this should be done like.

---

Which other packages are you referring to? Any examples?

---

Further, is there any official conventions on this?

---

> Which other packages are you referring to? Any examples?

Some package that does it like this https://github.com/NixOS/nixpkgs/blob/4a8ed1661dcc22f90f24c80c9ab1f88a43d71563/pkgs/tools/misc/duf/default.nix#L18

> Further, is there any official conventions on this?

If there is a flag or flagsArray use this because usually it does way more in the background which we do not want to overwrite.

---

Got it. Fixed :+1: